### PR TITLE
Remove not necessary type forwarded from attributes when type comes from mscorlib in netfx

### PR DESCRIPTION
**Pull request by safern**
cc: @ViktorHofer @danmosemsft @ericstj @jkotas

### CODE REVIEW

**Comment by jkotas** [comment]
Does this have an impact on other (non-binary formatter) serializers?

**Comment by danmosemsft** [comment]
@zhenlan for DCS

**Comment by zhenlan** [comment]
@huanwu @Lxiamail

**Comment by Lxiamail** [comment]
@safern Can you give more context why we make this change? Why the typeforwarder was there in the first place?

**Comment by ViktorHofer** [comment]
I added the typeforwards with https://github.com/dotnet/coreclr/pull/12300 (ignore that the commit is from the release branch, was easier to dig up). We later found out that System.Private.CoreLib is specialized and doesn't need the typeforwards. Only internal types need the type forward.

File: src/System.Private.CoreLib/src/System/AppDomainUnloadedException.cs [hunk 1]
```diff
@@ -17,7 +17,6 @@
 namespace System
 {
     [Serializable]
-    [System.Runtime.CompilerServices.TypeForwardedFrom("mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
```
Comment (ViktorHofer) [inline src/System.Private.CoreLib/src/System/AppDomainUnloadedException.cs]: internal types still need the type forward

**Comment by ViktorHofer** [comment]
As discussed offline. Thanks for doing this!

**Pull request merged**